builtin/logical/ssh/path_sign.go

```diff
@@ -172,12 +213,28 @@ func (b *backend) pathSignCertificate(ctx context.Context, req *logical.Request,
 		return nil, errwrap.Wrapf("failed to parse stored CA private key: {{err}}", err)
 	}
 
+	var certValidAfter, certValidBefore uint64
+
+	// branch the logic for calculating the certificate ValidAfter & ValidBefore values
+	if validAfterWasSpecified && validBeforeWasSpecified {
+		// the ValidAfter/ValidBefore were specified as epoch's directly in the specified data
+		// it's already known that these are both true if at least one is true from prior checks
+		certValidAfter = validAfter
+		certValidBefore = validBefore
+	} else {
+		// calculate certificate ValidAfter/ValidBefore using TTL (specified or derived)
```

Comment:
The default conditional logic here is to do the previous behavior (use the TTL [specified or derived] to calculate the ValidAfter/ValidBefore values)